src/utils/breakpoint/astBreakpointLocation.js

```diff
@@ -27,15 +27,22 @@ export function getASTLocation(
     const line = location.line - scope.location.start.line;
     return {
       name: scope.name,
+      signature: scope.signature,
       offset: { line, column: undefined }
     };
   }
-  return { name: undefined, offset: location };
+  return { name: undefined, signature: undefined, offset: location };
 }
 
-export async function findScopeByName(source: Source, name: ?string) {
+export async function findScopeByName(
+  source: Source,
+  name: ?string,
+  signature: ?string
+) {
   const symbols = await getSymbols(source.id);
   const functions = symbols.functions;
 
-  return functions.find(node => node.name === name);
+  return functions.find(
+    node => node.name === name && node.signature === signature
```

Comment:
@loganfsmyth In answer to your question on bail out of `signatures`, Yes! if the `signature` is not unique, then the result of the check of `name` and `signature` would match that of checking just the `name`.